ufl/core/terminal.py

```diff
@@ -102,3 +102,9 @@ class FormArgument(Terminal):
 
     def __init__(self):
         Terminal.__init__(self)
+
+    def ufl_strip_data(self):
+        """Return a copy of the form argument with all non-UFL information
+        removed. This is useful for removing references to large data structures
+        so that the object is cachable."""
+        raise NotImplementedError("ufl_strip_data should be implemented by the subclass")
```

Comment:
I think I would prefer if these removal of data just lived as a handler on the transformer.

Reply:
Yes that would definitely be neater.

Reply:
This is now done.